src/compute/flexbox.rs

```diff
@@ -1733,24 +1737,43 @@ fn calculate_flex_item(
 
     if direction.is_row() {
         let baseline_offset_cross = total_offset_cross + item.offset_cross + item.margin.cross_start(direction);
-        let inner_baseline = preliminary_size_and_baselines.first_baselines.y.unwrap_or(preliminary_size.height);
+        let inner_baseline = layout_output.first_baselines.y.unwrap_or(size.height);
         item.baseline = baseline_offset_cross + inner_baseline;
     } else {
         let baseline_offset_main = *total_offset_main + item.offset_main + item.margin.main_start(direction);
-        let inner_baseline = preliminary_size_and_baselines.first_baselines.y.unwrap_or(preliminary_size.height);
+        let inner_baseline = layout_output.first_baselines.y.unwrap_or(size.height);
         item.baseline = baseline_offset_main + inner_baseline;
     }
 
-    *tree.get_unrounded_layout_mut(item.node) = Layout {
-        order: item.order,
-        size: preliminary_size_and_baselines.size,
-        location: Point {
-            x: if direction.is_row() { offset_main } else { offset_cross },
-            y: if direction.is_column() { offset_main } else { offset_cross },
-        },
+    let location = Point {
+        x: if direction.is_row() { offset_main } else { offset_cross },
+        y: if direction.is_column() { offset_main } else { offset_cross },
     };
 
-    *total_offset_main += item.offset_main + item.margin.main_axis_sum(direction) + preliminary_size.main(direction);
+    *tree.get_unrounded_layout_mut(item.node) = Layout { order: item.order, size, content_size, location };
+
+    *total_offset_main += item.offset_main + item.margin.main_axis_sum(direction) + size.main(direction);
+
+    #[cfg(feature = "content_size")]
+    {
+        let size_content_size_contribution = Size {
+            width: match item.overflow.x {
+                Overflow::Visible => f32_max(size.width, content_size.width),
+                _ => size.width,
+            },
+            height: match item.overflow.y {
+                Overflow::Visible => f32_max(size.height, content_size.height),
+                _ => size.height,
+            },
+        };
+        if size_content_size_contribution.width > 0.0 && size_content_size_contribution.height > 0.0 {
```

Comment:
I can see this > 0.0 check 3 times in this PR, perhaps it warrants a helper function on `Size`